drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -53,6 +57,12 @@ static struct nrf5_802154_data nrf5_data;
 #define FRAME_PENDING_BYTE 1
 #define FRAME_PENDING_BIT (1 << 4)
 
+#if defined(CONFIG_SOC_NRF5340_CPUAPP) || defined(CONFIG_SOC_NRF5340_CPUNET)
```

Comment:
Does the driver software execute on the nRF5340 application core?
Then, does it run on the non-secure domain? If so, I believe the FICR registers are not directly accessible.
Can you clarify?

Reply:
There are two modules: 
1. 802.15.4 radio driver that is executed in the network core
2. 802.15.4 zephyr shim that is executed in the application core

The radio driver does not use any hardware assigned eui64 addresses. The higher layer is responsible to manage addresses for the driver. However, the shim layer has eui64 in its API. The code you commented is taken from the shim and it is used to create eui64. It is executed in the application core.

I've tested some OpenThread examples and it looks that the shim layer works as expected. I'm not sure if these examples use trustzone or eui64.

Because support for 802.15.4 for nRF53 is for development now, I think we can keep this implementation with a known limitation and fix it right after the upmerge.

Thoughts?

Reply:
Upstream OpenThread probably run in secure mode. That's why you cannot see the issues.
As it is now, this 802.15.4 zephyr shim layer cannot run in Non-Secure mode. Probably won't compile either.

Can you put some #ifndef CONFIG_TRUSTED_EXECUTION_NONSECURE around the FICR access? (clearly, for nRF5340 CPUAPP only).

Reply:
@ioannisg , do you have any suggestions how this code could be modified to run in non secure mode? Is there some kind of service to read parts of ficr from non secure environment? I'm asking to plan to fix this in a subsequent pr.

Reply:
Yes, the Nordic SDK contains secure services to read parts of FICR (or in general, secure memory). There are sample that showcase the usage.
For now, I would suggest to just do not allow the driver to be built for and run in Non-Secure domain, since direct access to FICR is not possible.